### PR TITLE
Sharing: add new Skype sharing button

### LINKED ISSUE #3170: Add Skype share button to Jetpack
Hello, I am following up on a twitter request by one of your users. We recently launched the Skype Share button. The configuration page is available here: https://www.skype.com/en/create-share-buttons/

We offer a JS button with SDK. We also offer a raw URL option. The URL is: 
https://web.skype.com/share?url=<urlencoded_url_to_share>&lang=en-US&flow_id=<GUID>&source=JetPack

Expected usage, with ideal window size: 
function shareUsingSkypeRawLink() {
window.open('https://web.skype.com/share?url=<urlencoded_url_to_share>&lang=en-US&flow_id=<GUID>&source=JetPack’, ‘_blank', 'toolbar=no, scrollbars=yes, resizable=yes, width=305, height=665');
              return false;
}
 <a href=‘javascript:void(0)' onclick="shareUsingSkypeRawLink()"> Share using link </a> 

Asset available here: 
Skype logo file is available here: https://swx.cdn.skype.com/skypewebsdk/shareButton/v/latest/assets/images/s_logo.svg

We also have a WP plugin: https://wordpress.org/plugins/skype-share-button/

Please let me know if you need any other information. 
Thanks, 
Amit Paunikar
Product@Skype

### PR DESCRIPTION
**Pull request by jeherve**
Fixes #3170

You'll need to run `grunt` to see the button styles in the dashboard.

### CODE REVIEW

File: modules/sharedaddy/sharing-sources.php [hunk 1]
```diff
@@ -1601,3 +1601,88 @@ function jetpack_sharing_pocket_init() {
...
+
+class Share_Skype extends Sharing_Source {
+	public $shortname = 'skype';
+	public $genericon = '\f220';
+	private $share_type = 'default';
+
+	public function __construct( $id, array $settings ) {
+		parent::__construct( $id, $settings );
+
+		if ( isset( $settings['share_type'] ) )
```
Comment (zinigor) [inline modules/sharedaddy/sharing-sources.php]: Please add curly braces in the conditionals here.

File: modules/sharedaddy/sharing-sources.php [hunk 2]
```diff
@@ -1601,3 +1601,88 @@ function jetpack_sharing_pocket_init() {
...
+			$this->smart = true;
+		else
+			$this->smart = false;
+	}
+
+	public function get_name() {
+		return __( 'Skype', 'jetpack' );
+	}
+
+	public function display_header() {
```
Comment (zinigor) [inline modules/sharedaddy/sharing-sources.php]: Let's get rid of the empty method, the service works fine without it.

**Comment by zinigor** [comment]
Left two minor comments, otherwise this works great, thanks!

**Comment by dereksmart** [comment]
:shipit: nice

**Pull request merged**